src/CommunityToolkit.Mvvm.SourceGenerators/Extensions/MethodDeclarationSyntaxExtensions.cs

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace CommunityToolkit.Mvvm.SourceGenerators.Extensions;
+
+/// <summary>
+/// Extension methods for the <see cref="MethodDeclarationSyntax"/> type.
+/// </summary>
+internal static class MethodDeclarationSyntaxExtensions
+{
+    /// <summary>
+    /// Checks whether a given <see cref="MethodDeclarationSyntax"/> has or could potentially have any attribute lists.
+    /// </summary>
+    /// <param name="methodDeclaration">The input <see cref="MethodDeclarationSyntax"/> to check.</param>
+    /// <returns>Whether <paramref name="methodDeclaration"/> has or potentially has any attribute lists.</returns>
+    public static bool HasOrPotentiallyHasAttributeLists(this MethodDeclarationSyntax methodDeclaration)
+    {
+        // If the declaration has any attribute lists, there's nothing left to do
+        if (methodDeclaration.AttributeLists.Count > 0)
+        {
+            return true;
+        }
+
+        // If there are no attributes, check whether the method declaration has the partial keyword. If it
+        // does, there could potentially be attribute lists on the other partial definition/implementation.
+        foreach (SyntaxToken modifier in methodDeclaration.Modifiers)
+        {
+            if (modifier.IsKind(SyntaxKind.PartialKeyword))
+            {
+                return true;
+            }
+        }
+
+        return false;
```

Comment:
Roslyn has an `Any` extension method.

```suggestion
        return methodDeclaration.Modifiers.Any(SyntaxKind.PartialKeyword);
```

Reply:
Ooh I didn't know, very nice! Thank you! 😄
Fixed in https://github.com/CommunityToolkit/dotnet/pull/633/commits/66f582bd94a8bd922661ae9ea960fd144fb232ca.